dpnp/dpnp_algo/dpnp_algo_indexing.pyx

```diff
@@ -65,12 +64,18 @@ ctypedef void(*custom_indexing_6in_func_ptr_t)(void *, void * , void * , const s
 ctypedef void(*fptr_dpnp_nonzero_t)(const void * , void * , const size_t, const size_t * , const size_t , const size_t)
 
 
-cpdef utils.dpnp_descriptor dpnp_choose(object input, list choices):
-    cdef shape_type_c obj_shape = utils._object_to_tuple(len(input))
-    cdef utils.dpnp_descriptor res_array = utils_py.create_output_descriptor_py(obj_shape, choices[0].dtype, None)
+cpdef utils.dpnp_descriptor dpnp_choose(dpnp_descriptor input, dpnp_descriptor choices):
+    cdef utils.dpnp_descriptor res_array = utils_py.create_output_descriptor_py(input.shape, choices[0].dtype, None)
```

Comment:
Let's use `utils.create_output_descriptor` instead of `utils_py.create_output_descriptor_py`.